modules/light-clients/07-tendermint/migrations.go

```diff
@@ -0,0 +1,72 @@
+package tendermint
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	host "github.com/cosmos/ibc-go/v6/modules/core/24-host"
+	"github.com/cosmos/ibc-go/v6/modules/core/exported"
+)
+
+// PruneTendermintConsensusStates prunes all expired tendermint consensus states. This function
+// may optionally be called during in-place store migrations. The ibc store key must be provided.
+func PruneTendermintConsensusStates(ctx sdk.Context, cdc codec.BinaryCodec, storeKey storetypes.StoreKey) error {
+	store := ctx.KVStore(storeKey)
+
+	// iterate over ibc store with prefix: clients/07-tendermint,
+	tendermintClientPrefix := []byte(fmt.Sprintf("%s/%s", host.KeyClientStorePrefix, exported.Tendermint))
+	iterator := sdk.KVStorePrefixIterator(store, tendermintClientPrefix)
+
+	var clients []string
+
+	// collect all clients to avoid performing store state changes during iteration
+	defer iterator.Close()
```

Comment:
we can potentially ignore an error here, is this a big deal?

Reply:
good question. This error is ignored in all our iterator functions. Looks like the SDK recently added a function to handle this https://github.com/cosmos/cosmos-sdk/issues/11785. Looks like the change hasn't been released yet. Should I update this function to handle the error returned or are we comfortable adding the `LogDeferred` usage when it is made available to us?

Reply:
Happy to wait for LogDeffered!